src/ecanvasser/services/survey.service.ts

```diff
@@ -0,0 +1,279 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common'
+import { HttpService } from '@nestjs/axios'
+import { EcanvasserService } from './ecanvasser.service'
+import { EcanvasserApiUtil } from '../util/ecanvasser.util'
+import {
+  ApiEcanvasserSurvey,
+  ApiEcanvasserSurveyQuestion,
+  ApiEcanvasserTeam,
+} from '../ecanvasser.types'
+import { CreateSurveyDto } from '../dto/create-survey.dto'
+import { CreateSurveyQuestionDto } from '../dto/create-survey-question.dto'
+import { UpdateSurveyQuestionDto } from '../dto/update-survey-question.dto'
+import { UpdateSurveyDto } from '../dto/update-survey.dto'
+
+@Injectable()
+export class SurveyService {
+  private readonly logger = new Logger(SurveyService.name)
+  private readonly apiUtil: EcanvasserApiUtil
+
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly ecanvasserService: EcanvasserService,
+  ) {
+    this.apiUtil = new EcanvasserApiUtil(httpService)
```

Comment:
This is an antipattern in Nest or DI pattern systems in general.

If `EcanvasserService` was an actual service that interacts/implements the integration w/ the Ecanvasser API, then Nest would be providing a _singleton_ in memory for this service to then consume and interact with. No need to create and manage a separate instance here as a private member on _this_ service.